app/templates/components/add-cron-job.hbs

```diff
@@ -1,32 +1,70 @@
-<form {{action (perform save) on="submit"}}>
-  <div class="form-elem select">
-    <span class="label">Branch</span>
-    {{#x-select value=selectedBranch required="true" on-change=(action (mut selectedBranch)) as |xs|}}
-      {{#each branches as |branch|}}
-        {{#xs.option value=branch}}{{branch.name}}{{/xs.option}}
-      {{/each}}
-    {{/x-select}}
+{{#travis-form 
+  class='form--cron' 
+  onSubmit=(action 'handleSubmit') as |form| }}
+  <div class='form-elem select'>
+    {{#form.field
+      class='branch-selector'
+      label='Branch'
+      requiresValidation=false
+      required=true
+      value=selectedBranch
+      onChange=(action (mut selectedBranch))
+      as |field|
+    }}
+      {{#field.select 
+        searchField='name' 
+        placeholder='Type branch name'
+        oninput=(action 'onQueryChange') 
+        options=comparedBranches as |branch|}}
+        {{branch.name}}
+      {{else}}
+        <div class='ember-power-select-option ember-power-select-option--no-matches-message'>
+          {{#if branchesFetcher.loadingData}}
+            Loading...
+            {{else}}
+            No results found
+          {{/if}}
+        </div>
+      {{/field.select}}
+    {{/form.field}}
   </div>
-  <div class="form-elem select">
-    <span class="label">Interval</span>
-    {{#x-select value=selectedInterval required="true" on-change=(action (mut selectedInterval)) as |xs|}}
-      {{#each intervals as |interval|}}
-        {{#xs.option value=interval}}{{interval}}{{/xs.option}}
-      {{/each}}
-    {{/x-select}}
+  <div class='form-elem select'>
+    {{#form.field
+      label='Interval'
+      value=selectedInterval
+      requiresValidation=false
+      required=true
+      onChange=(action (mut selectedInterval))
+      as |field|
+    }}
+      {{#field.select options=intervals as |interval|}}
+        {{interval}}
+      {{/field.select}}
+    {{/form.field}}
   </div>
-  <div class="form-elem select">
-    <span class="label">Options</span>
-    {{#x-select value=selectedOption required="false" on-change=(action (mut selectedOption)) as |xs|}}
-      {{#xs.option value=false}}{{options.[0]}}{{/xs.option}}
-      {{#xs.option value=true}}{{options.[1]}}{{/xs.option}}
-    {{/x-select}}
+  <div class='form-elem select'>
+    {{#form.field
+      label='Options'
+      value=selectedOption 
+      requiresValidation=false
+      required=true
+      onChange=(action (mut selectedOption))
+      as |field|
+    }}
+      {{#field.select options=options as |option|}}
+        {{option}}
+      {{/field.select}}
+    {{/form.field}}
   </div>
-  <div class="form-elem">
-    {{#if save.isRunning}}
-      {{loading-indicator}}
-    {{else}}
-      <input type="submit" value="Add" class="form-submit" />
-    {{/if}}
+  <div class='form-elem'>
+    <button
+      onclick={{form.submit}}
+      class='form-submit'>
+      {{#if save.isRunning}}
```

Comment:
You have removed the `save` task from component, because of which now it adds a cron row immediately, without waiting for server response, and does not show loading indication.